### PR TITLE
Feature/freelancer create form

### PR DESCRIPTION
**Pull request by SusannaRiccardi**

### CODE REVIEW

File: routes/freelance/router.js [hunk 1]
```diff
@@ -236,13 +237,49 @@ router.put('/:freelanceid/availability', function(req, res, next) {
...
   newFreelance.save(function(err, saved) {
     if (err) {
       res.status(400).json(utils.formatErrorMessage(err));
+      return;
+    }
+    // Now we can parse, create and save the tags.
+    let tagObjects = [];
+    let count = 0;
+    if (tags != undefined) {
+      for (let tag of tags) {
```
Comment (aronfiechter) [inline routes/freelance/router.js]: What if one or more of the tags that the user entered already exists? If I understood correctly this method doesn't check that.

File: routes/freelance/router.js [hunk 2]
```diff
@@ -236,13 +237,49 @@ router.put('/:freelanceid/availability', function(req, res, next) {
...
+        const newTag = new Tag({
+          tagName: tag,
+        });
+        newTag.freelancers.push(saved);
+        newTag.save(function(errTag, savedTag) {
+          if (err) {
+            res.status(400).json(utils.formatErrorMessage(errTag));
+          }
+          saved.tags.push(savedTag);
```
Comment (aronfiechter) [inline routes/freelance/router.js]: If I recall correctly the tags of a `Freelance` are an `array` of `ObjectID`. Also, is this saved in the DB?

File: public/components/freelancer-create.js [hunk 3]
```diff
@@ -131,6 +130,30 @@ class CreationForm extends React.Component {
...
           <div className="group">
+            <input ref="description" className="job-description" name="job-descritpion" type="text" required/>
+            <span className="bar"></span>
+            <label>
+              Job Description
+            </label>
+          </div>
+
+          <div className="group">
+            <input ref="tags" className="job-tags" name="job-tags" type="text"/>
```
Comment (aronfiechter) [inline public/components/freelancer-create.js:141]: Maybe we should check that the tags are entered actually as comma separated tags. We could do it with a regex, like for the phone number.

**Review by aronfiechter — DISMISSED** [review]
Good job, but I'm not sure it's completely done on the server side. (For the client side part we could also just not care about wrong inputs though.)

File: routes/freelance/router.js (hunk 2 above)
Comment (BalestraPatrick) [reply routes/freelance/router.js]: Yes, they are saved correctly as we tested even in this way.

Comment (aronfiechter) [reply routes/freelance/router.js]: But you add something to `saved.tags`, but then you don't save `saved` again.

Comment (BalestraPatrick) [reply routes/freelance/router.js]: We save it in `sendResponse`.

**Pull request merged**